### PR TITLE
Update golang to 1.13

### PR DESCRIPTION
**Pull request by glower**
* Migrating to go mod
* Migrating to go  1.13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,11 +15,10 @@ before_install:
     - go get gopkg.in/retry.v1
     - go get gopkg.in/tomb.v2
     - go get gopkg.in/check.v1
-    - go get github.com/golang/lint/golint
+    - go get golang.org/x/lint/golint
     - go get github.com/client9/misspell/cmd/misspell
     - go get -u github.com/gordonklaus/ineffassign
     - go get -u github.com/alexkohler/nakedret
-    - sh ./get-deps.sh
```
Comment (glower) [inline .travis.yml]: go will manage dependencies now, so no need for extra tools

File: .travis.yml [hunk 2]
```diff
@@ -15,11 +15,10 @@ before_install:
     - go get gopkg.in/retry.v1
     - go get gopkg.in/tomb.v2
     - go get gopkg.in/check.v1
-    - go get github.com/golang/lint/golint
+    - go get golang.org/x/lint/golint
```
Comment (glower) [inline .travis.yml]: this linter was moved to the new repo

File: account/account.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ import (
 // FetchAssertionFromStore retrieves an assertion from the store
 var FetchAssertionFromStore = func(modelType *asserts.AssertionType, headers []string) (asserts.Assertion, error) {
 	var user *auth.UserState
-	var authContext auth.AuthContext
+	var authContext store.DeviceAndAuthContext
```
Comment (glower) [inline account/account.go]: version of snapd was updated and this package was moved

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,29 @@
+module github.com/CanonicalLtd/serial-vault
```
Comment (glower) [inline go.mod]: this is auto generated from `go mod init`

File: go.sum [hunk 5]
```diff
@@ -0,0 +1,78 @@
+github.com/boltdb/bolt v1.3.1 h1:JQmyP4ZBrce+ZQu0dY660FMfatumYDLun9hBCUVIkF4=
```
Comment (glower) [inline go.sum]: this is auto generated from `go mod init`

File: run-tests.sh [hunk 6]
```diff
@@ -16,7 +16,7 @@
 
 set -e
 
-TESTS_EXTRAS_URL="https://git.launchpad.net/~snappy-hwe-team/snappy-hwe-snaps/+git/tests-extras"
```
Comment (glower) [inline run-tests.sh]: don't know if this is used, but the tool was moved to the new repo, the old one is deprecated

**Comment by glower** [comment]
* vendor/vendor.json is now replaced by go.mod and go.sum (auto generated)

**Comment by glower** [comment]
update for docker and snap build will follow in the next PR

File: account/account.go (hunk 3 above)
Comment (doismellburning) [reply account/account.go]: I don't love that this PR includes a stealth dependency update - was the snapd dependency not pinned previously?

Comment (glower) [reply account/account.go]: Yes, it was, now the version was generated by `go mod init`. Let me try to pin the version to the previous one

Comment (glower) [reply account/account.go]: Hmm, can't pin the version of snapd because of this:
https://bugs.launchpad.net/snapd/+bug/1821959

Comment (doismellburning) [reply account/account.go]: Let's please not introduce unpinned dependencies - that is the path to sadness and woe and surprise breakage and lack of reproducibility

**Comment by glower** [comment]
Because we can't pin to the previous version of the snapd we will postpone the update

**Comment by glower** [comment]
Because we can't pin to the previous version of the snapd we will postpone the update

**Pull request closed without merging**